addon/components/ember-table.js

```diff
@@ -81,29 +81,34 @@ StyleBindingsMixin, ResizeHandlerMixin, {
 
   // An array of the rows currently selected. If `selectionMode` is set to
   // 'single', the array will contain either one or zero elements.
-  selection: Ember.computed(function(key, val) {
-    var selectionMode = this.get('selectionMode');
-    if (arguments.length > 1 && val) {
-      this.get('persistedSelection').clear();
-      this.get('rangeSelection').clear();
+  selection: Ember.computed('persistedSelection.[]', 'rangeSelection.[]', 'selectionMode', {
+    set: function(key, val) {
+      if (val) {
+        var selectionMode = this.get('selectionMode');
+        this.get('persistedSelection').clear();
+        this.get('rangeSelection').clear();
+        switch (selectionMode) {
+          case 'single':
+            this.get('persistedSelection').addObject(val);
+            break;
+          case 'multiple':
+            this.get('persistedSelection').addObjects(val);
+        }
+      }
```

Comment:
for consistency, this should `return this.get('selection')` or duplicate the code.